frontend/src/components/StatusPage/StatusPageDetailData.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { formatDistanceToNow, parseISO } from "date-fns";
+import { Address, Status, Part } from "solo-types";
+
+interface StatusPageDetailDataProps {
+  address: Address[];
+  status: Status[];
+  part: Part[];
+}
+
+const StatusPageDetailData: React.FC<StatusPageDetailDataProps> = ({
+  address,
+  status,
+  part
+}) => {
+  const shipperAddy = address.find(addy => addy.address_type === "2")?.name;
+  const receiverAddy = address.find(addy => addy.address_type === "3")?.name;
```

Comment:
I don't love the need to do a `.find` on every render. My first recommendation would be to only pass in the address needed and make the component dumber.  The hard coded address_type feels weird. If that isn't an option (I'd like to know why) at least memoize it (`useMemo`)

Reply:
Ultimately what I would do is just hold this data as an object and look up by key.

Reply:
Added explicit conversion step from the structure of the api response to a more frontend focused data structure.